src/components/hooks/useBookmarks.js

```diff
@@ -288,69 +287,102 @@ const useBookmarkMock = () => {
+  const relevantWorksByBookmarkId = bookmarks?.map((bookmark) => {
+    const materialTypes = bookmark?.materialType?.split(" / ");
+    const work = workByIdsDataRemovedDuplicates?.find(
+      (w) => w?.workId === bookmark?.workId
+    );
 
-  const { data: workByPidsData, isLoading: pidsToWorkLoading } = useData(
-    workFragments.pidsToWorks({
-      pids: workPids?.map((work) => work.materialId),
-    })
-  );
+    let manifestationWithCorrectMaterialType =
+      work?.manifestations?.mostRelevant.filter((m) =>
+        isMaterialTypesMatch(materialTypes, m?.materialTypes)
+      );
 
-  // Reorganize order and add bookmark data
-  const data = useMemo(() => {
-    if (!bookmarks) return [];
-    const transformedWorkByIds = workByIdsData?.works?.map((work) => ({
-      ...work,
-      materialId: work?.workId,
-    }));
-    const transformedWorkByPids = workByPidsData?.manifestations?.map(
-      (work) => ({
-        ...work,
-        materialId: work?.pid,
-      })
+    // if bookmarkId is in specificEdition array, then filter the specific edition out
+    const specificEditionBookmark = specificEditions?.find(
+      (se) => se?.bookmarkId === bookmark?.bookmarkId
     );
-    const merged = [].concat(transformedWorkByIds, transformedWorkByPids);
-
-    return bookmarks
-      .map((bookmark) => {
-        const workData = merged.find(
-          (item) => item?.materialId === bookmark.materialId
+    if (specificEditionBookmark) {
+      const specificManifestation =
+        manifestationWithCorrectMaterialType?.filter(
+          (m) => m?.pid === specificEditionBookmark?.materialId
         );
-        if (!workData) {
-          return null;
+      manifestationWithCorrectMaterialType = specificManifestation;
+    }
+    return {
+      ...work,
+      bookmarkId: bookmark?.bookmarkId,
+      materialId: bookmark?.materialId,
+      pid: specificEditionBookmark ? bookmark?.materialId : undefined,
+      key: bookmark?.key,
+      workId: bookmark?.workId,
+      manifestations: manifestationWithCorrectMaterialType,
+    };
+  });
+
+  const data = useMemo(() => {
+    if (!bookmarks) return [];
+    //cannot group if there are multiple worktypes for same workid in order
+    const groupedByWorkIdAndType = {};
+    relevantWorksByBookmarkId?.forEach((rw) => {
+      rw.manifestations?.forEach((manifestation) => {
+        const key = rw?.key;
+        if (!groupedByWorkIdAndType[key]) {
+          groupedByWorkIdAndType[key] = [];
         }
+        groupedByWorkIdAndType[key].push(manifestation);
+      });
+    });
 
-        // Merge data
-        return {
-          ...workData,
-          ...bookmark,
-        };
-      })
-      .filter((item) => item); // filter nulls
-  }, [bookmarks, workByPidsData, workByIdsData]);
-  const isLoading = idsToWorksLoading || pidsToWorkLoading;
-  return { data, isLoading };
+    return relevantWorksByBookmarkId.filter((item) => item); // filter nulls
+  }, [bookmarks, workByIdsData]);
```

Comment:
Kan du forklare den her?

Hvorfor den kommentar? Vil vi ikke gerne kunne gruppere hvis der er flere materialeTyper (antager du mente materialtypes og ikke worktypes)

Reply:
den første 
"cannot group if there are multiple worktypes for same workid in order" er en gammle kommentar, den sletter jeg eller tifløjer noget, det var derfor at jeg tilføjede materialtype til nøglen.

Reply:
tror faktisk kan jeg flytte relevantWorksByBookmarkId ind i den her og behøver ikke sammle dem på den måde længere...